src/integration_tests/datasets/dataset_endpoints_integration_test.py

```diff
@@ -0,0 +1,135 @@
+        
+
+    @classmethod
+    def teardown_class(self) -> None:
+        cleanup()
+        inject_wait_time(3) # Inject wait time to allow all message to be processed
+
+    @pytest.mark.order(1)
+    def test_dataset_upload_and_metadata(self):
+        """
+        Test uploading and retriving dataset metadata.
+
+        - Sends a GET request to retrieve metadata for a dataset
+        - Verifies that the dataset was uploaded successfully
+        - Asserts the metadata retrieved matches the expected structure.
+        """
+
+        response = self.session.get(
+            f"{config.API_URL}/v1/dataset_metadata?survey_id=test_survey_id&period_id=test_period_id_2",
+            headers = self.headers
+        )
+
+        assert self.dataset is not None, "Dataset upload failed"
+        assert response.status_code == 200, "not found"
+
+        metadata_data = response.json()
+
+        assert metadata_data == [{
+        "dataset_id": "3",
+        "survey_id": "test_survey_id",
+        "period_id": "test_period_id_2",
+        "form_types": ["knj", "okn", "ojdw"],
+        "title": "Which side was better?",
+        "sds_published_at": "2023-04-20T12:00:00Z",
+        "total_reporting_units": 2,
+        "schema_version": "v1.0.0",
+        "sds_dataset_version": 1,
+        "filename": "test_filename.json",
+        } ]
+
+    @pytest.mark.order(2)
+    def test_grabbing_unit_data(self):
+        """
+        Test retrieving unit data for a dataset
+
+        - Get request to retrieve unit data for a dataset
+        - Asserts the status code, if its found (200) or not (404)
+        - Asserts if unit data matches the expected structure
+        """
+
+        response = self.session.get(
+            f"{config.API_URL}/v1/unit_data?dataset_id=0&identifier=43532",
+            headers = self.headers
+        )
+
+        assert response.status_code == 200, "not found"
+
+        unit_data = response.json()
+
+        assert unit_data == {
+        "dataset_id": "0",
+        "survey_id": "test_survey_id",
+        "period_id": "test_period_id",
+        "schema_version": "v1.0.0",
+        "form_types": ["jke", "als", "sma"],
+        "data": "test",
+        }
+
+    @pytest.mark.order(3)
+    def test_dataset_without_title(self):
+        """
+        Test retrieving a dataset metadata without a title
+
+        - Get request to retrieve a metadata missing a title
+        - Assert status code is 200
+        - Checks if the metadata matches the expected structure without a title
+        """
+
+        response = self.session.get(
+            f"{config.API_URL}/v1/dataset_metadata?survey_id=test_survey_id&period_id=test_period_id",
+            headers = self.headers
+        )
+
+        assert response.status_code == 200, "not found"
+
+        metadata_without_title = response.json()
+
+        assert metadata_without_title[1] == {
```

Comment:
why are you using metadata_without_title[1] and not the object you originally uploaded.